app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala

```diff
@@ -306,13 +310,15 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
     }
   }
 
-  def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean, backUrl: String, id: Option[IncomeSourceId], incomeSourceType: IncomeSourceType)
-                   (implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+  def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean, backUrl: String, maybeIncomeSourceIdHash: Option[IncomeSourceIdHash],
+                    incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_], hc: HeaderCarrier): Future[Result] = {
+
+    val incomeSourceIdMaybe: Option[IncomeSourceId] = compare(incomeSourceIdHash = maybeIncomeSourceIdHash)
 
     withIncomeSourcesFS {
       for {
         value <- if (incomeSourceType == SelfEmployment) {
-          getManageIncomeSourceViewModel(sources = sources, incomeSourceId = id.get, isAgent = isAgent)
+          getManageIncomeSourceViewModel(sources = sources, incomeSourceId = incomeSourceIdMaybe.getOrElse(mkIncomeSourceId("")), isAgent = isAgent)
```

Comment:
Yes. I thought adding validation was outside the scope of this ticket since we don't really know the constraints on income source id. With current implementation, getManageIncomeSourceViewModel returns an error when passed an empty string